tests/vmi_gpu_test.go

```diff
@@ -0,0 +1,134 @@
+package tests_test
+
+import (
+	"encoding/xml"
+	"strings"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/util"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+func parseDeviceAddress(addrString string) []string {
+	addrs := strings.Split(addrString, ",")
+	naddrs := len(addrs)
+	if naddrs > 0 {
+		if addrs[naddrs-1] == "" {
+			addrs = addrs[:naddrs-1]
+		}
+	}
+
+	for index, element := range addrs {
+		addrs[index] = strings.TrimSpace(element)
+	}
+	return addrs
+}
+
+func checkGPUDevice(vmi *v1.VirtualMachineInstance, gpuName, prompt string) {
+	cmdCheck := "lspci"
+	err := tests.CheckForTextExpecter(vmi, []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: prompt},
+		&expect.BSnd{S: cmdCheck},
+		&expect.BExp{R: prompt},
+		&expect.BSnd{S: "echo $?\n"},
```

Comment:
Hm, I don't see where this function actually validates that the device is present; it only executes lspci without any arguments and then checks that it didn't fail (which is effectively a no-op).

Reply:
Updated to execute lspci with arguements